npm/nameSpaceController.go

```diff
@@ -304,12 +268,18 @@ func (nsc *nameSpaceController) syncNameSpace(key string) error {
 	}
 
 	if nsObj.DeletionTimestamp != nil || nsObj.DeletionGracePeriodSeconds != nil {
-		return nsc.cleanDeletedNamespace(util.GetNSNameWithPrefix(nsObj.Name))
+		return nsc.cleanDeletedNamespace(cachedNsKey)
+	}
 
+	cachedNsObj, nsExists := nsc.npmNamespaceCache.nsMap[cachedNsKey]
```

Comment:
Since we are checking the nsObj existence here? why not just call from syncAddNameSpace, instead of going through syncUpdateNameSpace and re-checking the same conditions again on the obj which wont change. 

I like the original approach here, of filtering these kind of events in updateNamespace or addNamespace, that way we do not burden our worker go routine to have to filter events.

Reply:
As we discussed, conceptually this approach has multiple advantages while we need measured number.
1. Can concurrently run event handlers (e.g., add, update, clean) and reconcile function since event handler does not need a lock.
2. Duplicated events will be collapsed into one in workqueue. So, it will get advantages when bulky events come since anyway, it will check last event object.